llmc/cuda_utils.cuh

```diff
@@ -260,4 +260,13 @@ __device__ __forceinline__ void stochastic_rounding(float in, float *out, unsign
     *out = in; // dummy function for when floatX is float (FP32 mode)
 }
 
+// Add two (potentially low-precision) vectors of size `n` together using stochastic rounding
+template<class Td, class Ts>
+__global__ void vector_add(Td* dst, const Ts* src, size_t n, unsigned seed) {
```

Comment:
Not sure if the time for this kernel is noticeable or not, but if it is, could make it much faster by using Packed128 and doing "src[idx] = 0" instead of a separate memset afterwards.

Reply:
implemented the packed kernel; to keep things simple, that meant assuming that `dst` and `src` are of the same type, which is true for our current use-case.